File: Day-12-of-30-Days/Solution.cpp
// Day 12 of 30 Days 👾

#include <iostream>
#include <algorithm>
using namespace std;

bool isAnagram(string str1, string str2) {
    str1.erase(remove_if(str1.begin(), str1.end(), [](char c) { 
        return !isalnum(c); }),
    str1.end());

    // converting all characters to lowercase
    transform(str1.begin(), str1.end(), str1.begin(), ::tolower);

    str2.erase(remove_if(str2.begin(), str2.end(), [](char c) { 
        return !isalnum(c); }),
    str2.end());

    // converting all characters to lowercase
    transform(str2.begin(), str2.end(), str2.begin(), ::tolower);

    // sorting
    sort(str1.begin(), str1.end());
    sort(str2.begin(), str2.end());

    return str1 == str2;
}

// Examples
int main() {
    cout << isAnagram("hello", "world") << endl; // returns 0 means false
    cout << isAnagram("test", "test") << endl; // returns 1 means true
    return 0;
}

File: Day-12-of-30-Days/Solution.py
# Day 12 of 30 Days 👾
import re

def is_angram(str1, str2):
    s1 = re.sub("[^a-zA-Z]", "", str1).lower()
    s2 = re.sub("[^a-zA-Z]", "", str2).lower()

    text1 = sorted(s1)
    text2 = sorted(s2)

    return text1 == text2


## Examples
print(is_angram("test", "test")) # True
print(is_angram("hello", "world")) # False